test/unit/tools/unified_spec_runner.test.ts

```diff
@@ -37,12 +37,18 @@ describe('Unified Spec Runner', function () {
           it('throws AssertionError when it finds extra keys', function () {
             actual =
               '{"data": {"$numberLong": "100"}, "a": {"$numberInt": "10"}, "b": {"$numberInt": "100"}}';
-            expect(() => resultCheckSpy(actual, expected, entitiesMap, [])).to.throw(
+            expect(() => resultCheckSpy(actual, expected, entitiesMap, [], true)).to.throw(
               AssertionError,
               /object has more keys than expected/
             );
           });
 
+          it('does not throw when it finds extra keys but checking is disabled', function () {
+            actual =
+              '{"data": {"$numberLong": "100"}, "a": {"$numberInt": "10"}, "b": {"$numberInt": "100"}}';
+            resultCheckSpy(actual, expected, entitiesMap, [], false);
```

Comment:
```suggestion
            expect(() => resultCheckSpy(actual, expected, entitiesMap, [], false)).to.not.throw;
```

Reply:
Is it not already the case that a test will fail if a function throws?

Reply:
It will but it's best practice in my opinion to wrap and assert the exception, rather than let it potentially propagate through.

Reply:
I'm fine with being explicit about the assertion given the method returns void. I've updated.